iOS/DittoPOS/Data/DittoExtensions.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  DittoExtensions.swift
+//  DittoPOS
+//
+//  Created by Shunsuke Kondo on 2023/12/28.
+//  Copyright © 2023 DittoLive Incorporated. All rights reserved.
+//
+
+import Combine
+import DittoSwift
+
+typealias DittoQuery = (string: String, args: [String: Any?])
+
+protocol DittoDecodable {
+    init(value: [String: Any?])
+}
+
+// MARK: - Extensions of `execute`
+extension DittoStore {
+
+    // Emit with mapped objects as an array
+    func executePublisher<T: DittoDecodable>(query: String, arguments: Dictionary<String, Any?>? = [:], mapTo: T.Type) -> AnyPublisher<[T], Error> {
+        return Future { promise in
+            Task.init {
+                do {
+                    let result = try await self.execute(query: query, arguments: arguments)
+                    let items = result.items.compactMap { T(value: $0.value) }
+                    promise(.success(items))
+                } catch {
+                    promise(.failure(error))
+                }
+            }
+        }.eraseToAnyPublisher()
+    }
+
+    // Emit with a mapped object as a single value instead of an array
+    func executePublisher<T: DittoDecodable>(query: String, arguments: Dictionary<String, Any?>? = [:], mapTo: T.Type, onlyFirst: Bool) -> AnyPublisher<T?, Error> {
+        return Future { promise in
+            Task.init {
+                do {
+                    let result = try await self.execute(query: query, arguments: arguments)
+                    guard let first = result.items.first else { return promise(.success(nil)) }
+                    let item = T(value: first.value)
+                    promise(.success(item))
+                } catch {
+                    promise(.failure(error))
+                }
+            }
+        }.eraseToAnyPublisher()
+    }
+}
+
+// MARK: - Extensions of `registerObserver`
+extension DittoStore {
+
+    // Send mapped objects as an array
+    func observePublisher<T: DittoDecodable>(query: String, arguments: [String : Any?]? = nil, deliverOn queue: DispatchQueue = .main, mapTo: T.Type) -> AnyPublisher<[T], Error> {
```

Comment:
Are theses publishers being used? I couldn't any code that called them.

Reply:
I leave Ditto extensions as customers references so that they can just copy-paste into their code.